app/src/main/java/com/x8bit/bitwarden/data/platform/processor/BridgeServiceProcessorImpl.kt

```diff
@@ -2,22 +2,54 @@ package com.x8bit.bitwarden.data.platform.processor
 
 import android.content.Intent
 import android.os.Build
+import android.os.IInterface
+import android.os.RemoteCallbackList
 import com.bitwarden.bridge.IBridgeService
 import com.bitwarden.bridge.IBridgeServiceCallback
 import com.bitwarden.bridge.model.EncryptedAddTotpLoginItemData
+import com.bitwarden.bridge.model.SharedAccountData
 import com.bitwarden.bridge.model.SymmetricEncryptionKeyData
 import com.bitwarden.bridge.model.SymmetricEncryptionKeyFingerprintData
+import com.bitwarden.bridge.util.NATIVE_BRIDGE_SDK_VERSION
+import com.bitwarden.bridge.util.encrypt
+import com.bitwarden.bridge.util.toFingerprint
+import com.bitwarden.bridge.util.toSymmetricEncryptionKeyData
+import com.x8bit.bitwarden.data.auth.datasource.disk.AuthDiskSource
+import com.x8bit.bitwarden.data.auth.repository.AuthRepository
+import com.x8bit.bitwarden.data.platform.datasource.disk.SettingsDiskSource
 import com.x8bit.bitwarden.data.platform.manager.FeatureFlagManager
+import com.x8bit.bitwarden.data.platform.manager.dispatcher.DispatcherManager
 import com.x8bit.bitwarden.data.platform.manager.model.FlagKey
 import com.x8bit.bitwarden.data.platform.util.isBuildVersionBelow
+import com.x8bit.bitwarden.data.vault.datasource.disk.VaultDiskSource
+import com.x8bit.bitwarden.data.vault.datasource.sdk.VaultSdkSource
+import com.x8bit.bitwarden.data.vault.repository.VaultRepository
+import com.x8bit.bitwarden.data.vault.repository.model.VaultUnlockData
+import com.x8bit.bitwarden.data.vault.repository.model.VaultUnlockResult
+import com.x8bit.bitwarden.data.vault.repository.util.statusFor
+import com.x8bit.bitwarden.data.vault.repository.util.toEncryptedSdkCipher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
 
 /**
  * Default implementation of [BridgeServiceProcessor].
  */
+@Suppress("LongParameterList")
 class BridgeServiceProcessorImpl(
+    private val authDiskSource: AuthDiskSource,
+    private val authRepository: AuthRepository,
     private val featureFlagManager: FeatureFlagManager,
+    private val settingsDiskSource: SettingsDiskSource,
+    private val vaultDiskSource: VaultDiskSource,
+    private val vaultRepository: VaultRepository,
+    private val vaultSdkSource: VaultSdkSource,
```

Comment:
Processors shouldn't have direct access to these "sources", can we abstract this behind a `BridgeRepository` or something like that?